hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/DynamicConfigYamlGenerator.java

```diff
@@ -152,6 +167,13 @@ String generate(Config config, boolean isMask) {
         networkConfigYamlGenerator(root, config);
         advancedNetworkConfigYamlGenerator(root, config);
         dataConnectionYamlGenerator(root, config);
+        namespacesConfigGenerator(root, config);
+        metricsConfigGenerator(root, config);
+        splitBrainProtectionConfigsGenerator(root, config);
+
+        // Reset maskSensitiveFields to default
+        DynamicConfigYamlGenerator.maskSensitiveFields = DEFAULT_MASK_SENSITIVE_FIELDS;
```

Comment:
Why is this required?

Reply:
This method changes the static field `maskSensitiveFields` to the passed variable when executed, so it made sense to me to return the state to how it was prior once we were finished - happy to remove.

Reply:
I think you are potentially fixing a bug here. If some code called `generate` with `mask == true` (didn't find any in production but still..), then any subsequent invocation of any other method in this class would generate masked values - including the config persistence feature which would result in a config file persisted with masked license key -> won't work on next restart.

Reply:
Restored in https://github.com/vbekiaris/hazelcast/commit/9f686c70bada9c576b0d8d8b149cf3d429c9ab80